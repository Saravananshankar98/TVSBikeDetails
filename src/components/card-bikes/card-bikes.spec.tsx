import { fireEvent, render } from "@testing-library/react";
import { TestContextProvider } from "../../test/test-container";
import CardBikes from "./card-bikes";

const vehicleFilterDetails = [ 
    {
      BikeName: "TVS Apache RTR 200",
      Price: "1.40 -1.45",
      CC: "197.75",
      PS: "20.82",
      nm: "17.25",
      RPM: "7500",
      KMPH: "35",
      Reviews: "281",
    }
  ];

describe("CardBikes", () => {
  let baseElement: any, container, screen;
  beforeEach(() => {
    ({ container, baseElement, ...screen } = render(
      <CardBikes vehicleFilterDetails={vehicleFilterDetails} />,
      {
        wrapper: TestContextProvider,
      }
    ));
  });

  it("Matches Snapshot", () => {
    expect(baseElement).toMatchSnapshot();
  });

  // it("click Know more options", async () => {
  //   const knowMoreButton = container.querySelectorAll(".MuiTypography-root");
  //   expect(knowMoreButton[0].textContent.trim()).toEqual("Scooters");
  //   fireEvent.click(knowMoreButton[0]);
  // });
});
