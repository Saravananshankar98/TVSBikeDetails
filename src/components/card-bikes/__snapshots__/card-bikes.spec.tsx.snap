// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBikes Matches Snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-2 css-r2d2jo-MuiGrid-root"
    />
  </div>
</body>
`;
